spec/rester/service/middleware/request_handler_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'support/test_service'
+
+module Rester
+  module Service::Middleware
+    RSpec.describe RequestHandler do
+      let(:service) {
+        Class.new(TestService).tap { |klass|
+          allow(klass).to receive(:name) { 'TestService' }
+        }
+      }
+      let(:id) { SecureRandom.uuid }
+      let(:env) do
+        {
+          'REQUEST_METHOD' => 'GET',
+          'PATH_INFO'      => '/v1/tests',
+          'CONTENT_TYPE'   => 'application/x-www-form-urlencoded',
+          'QUERY_STRING'   => '',
+          'rack.input'     => StringIO.new(''),
+          'HTTP_X_RESTER_CORRELATION_ID' => id,
+          'HTTP_X_RESTER_CONSUMER_NAME' => 'TestClient'
+        }
+      end
+
+      subject { service.call(env) }
+
+      it 'should set the producer name in the response' do
+        expect(subject[1]['HTTP_X_RESTER_PRODUCER_NAME']).to eq 'TestService'
+      end
```

Comment:
I don't think this is the right header format here.

If we curl, the header should be in the `X-Rester` format.